subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -125,4 +127,46 @@ public boolean isAllowUntrustedServer() {
     public void setAllowUntrustedServer(boolean allowUntrustedServer) {
         this.allowUntrustedServer = allowUntrustedServer;
     }
+
+    /**
+     * Specifies whether it is acceptable to communicate with a build cache over an insecure HTTP connection.
+     * <p>
+     * For security purposes this intentionally requires a user to opt-in to using insecure protocols on case by case basis.
+     * <p>
+     * Gradle intentionally does not offer a global system/gradle property that allows a universal disable of this check.
+     * <p>
+     * <b>Allowing communication over insecure protocols allows for a man-in-the-middle to impersonate the intended server,
+     * and gives an attacker the ability to
+     * <a href="https://max.computer/blog/how-to-take-over-the-computer-of-any-java-or-clojure-or-scala-developer/">
+     *     serve malicious executable code onto the system.
+     * </a>
+     * </b>
+     *
+     * @since 5.6
+     */
+    @Incubating
+    public boolean isAllowUntrustedProtocol() {
```

Comment:
`isAllow` sounds weird (I know it looks better in Groovy/Kotlin), but can we make it look okay in Java as well?

Reply:
@paplorinc Do you have a suggestion here for a better name?
Using the `get`/`set` standard here doesn't feel right either.

Reply:
Not sure, this always confuses me.

Reply:
I'd call it `isUntrustedProtocolAllowed` instead.